workflows/cloudify_system_workflows/snapshots/credentials.py

```diff
@@ -53,9 +58,10 @@ def dump(self, tempdir):
 
         for deployment_id, n in self._get_hosts():
             props = n.properties
-            if 'cloudify_agent' in props and 'key' in props['cloudify_agent']:
+            agent_config = self._get_agent_config(props)
+            if 'key' in agent_config:
                 node_id = deployment_id + '_' + n.id
-                agent_key_path = props['cloudify_agent']['key']
+                agent_key_path = agent_config['key']
```

Comment:
Good point, the whole thing will fall over if it hits one of those right now. I'll fix